src/jit/morph.cpp

```diff
@@ -11249,6 +11249,7 @@ GenTreePtr Compiler::fgMorphSmpOp(GenTreePtr tree, MorphAddrContext* mac)
             if (oper == GT_ADDR && (op1->gtOper == GT_LCL_VAR || op1->gtOper == GT_CLS_VAR))
             {
                 tree->gtFlags &= ~GTF_GLOB_REF;
+                op1->gtFlags &= ~GTF_GLOB_REF;
```

Comment:
That makes sense but I would try to find the change that added the comment that explicitly mentions that the flag is needed for the child. If it turns out that the IR invariants have changed since then, we should simply delete the comment as obsolete.